hibernate-core/src/main/java/org/hibernate/query/sqm/tree/select/SqmSelectClause.java

```diff
@@ -25,6 +27,7 @@
 public class SqmSelectClause extends AbstractSqmNode implements SqmAliasedExpressionContainer<SqmSelection>, JpaSelection<Object> {
 	private boolean distinct;
 	private List<SqmSelection<?>> selections;
+	private Set<String> aliases;
```

Comment:
IMO this set of aliases should be built on demand when a selection is added/set. We can replace all calls to `add` or `addSelection` that we currently have with some form of `setSelections` that also does this duplicate check, but over the full list of selections and only once. Also, the check should IMO go deeper, as I think that e.g. `DynamicInstantiation` and nested (not sure if possible) `SqmJpaCompoundSelection` are currently not handled. IMO the duplicate check logic should be moved to a utility class and be used in `SqmDynamicInstantiation`, `SqmJpaCompoundSelection` and `SqmSelectClause`.